### DISCUSSION TITLE
no decimal in rt of data through Firefox

### DISCUSSION THREAD

**Discussion by YukiTanida** [jspsych/jsPsych#834] [Q&A] [answered]
Hello, 

I have a question about "rt" data. When I get data through Firefox, rt is recorded without any decimals (e.g., 2746). On the other hand, rt generated through other browsers (e.g., safari, edge, chrome) have some decimals (e.g., 12811.789999948815). I know that this rt is in millisecond and so the Firefox's rt can be recoginsed to be strict enough. But it's uncomfortable. Does anyone know about or faced with this phenomenon?

I faced with this phenomenon regardless of plugin (e.g., survey-text, jspsych-html-keyboard-response, etc) and regardless of OS (macOS Mojave & Windows10). Therefore I think this phenomenon depends specifically on firefox. 

Thank you!

**Comment by vijaymarupudi** [comment, marked as answer]
Browsers manually fuzz the provided time for security reasons. If one knew the exact time, many memory attacks that allow reading passwords from other applications become possible. Firefox and Chrome use different fuzzing techniques, mentioned here: https://github.com/w3c/hr-time/issues/56

You can be sure that the time, regardless of browser, will be precise to plus or minus 2 ms.

**Reply by YukiTanida** [reply]
Thank you very much for replying!

I understand circumstances involving web-browser development. I was relieved to hear that, it's not my jsPsych code fault. In my case, precision with plus or minus 2ms noise is enough strict. I can start my experiment with no worries.

Thank you,
Yuki
